pkg/ccl/backupccl/restore_planning.go

```diff
@@ -2224,10 +2254,27 @@ func planDatabaseModifiersForRestore(
 			),
 		)
 
+		var maxSeenID descpb.ID
+		for _, desc := range sqlDescs {
+			if desc.GetID() > maxSeenID {
+				maxSeenID = desc.GetID()
+			}
+		}
 		// Allocate the region enum ID.
-		regionEnumID := maxSeenID + 1
-		regionEnumArrayID := maxSeenID + 2
-		maxSeenID += 2
+		regionEnumID, err := catalogkv.GenerateUniqueIDGreaterThanID(
```

Comment:
ditto comment as above, can we revert this change if it is not required? If it is, can you explain what changed please?

Reply:
Done